### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by AhmedMohsen600**
Screen Shot 2023-02-09 at 2 46 06 PM

in this branch we:

- Created List_service class to own all list methods 
- Created Get_service class to own all get methods
- Remove All the methods from app.rb file to List_service class and Get_service class and created instance from these classes

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+require_relative 'student'
+require_relative 'rental'
+require_relative 'teacher'
+
+def main
+  lib = App.new
+  lib.run
+end
+
+main
```
Comment (bobb-Rob) [inline main.rb:13]: - [ ] Well done implementing this project so far. I tried to run your project but I get this 👇🏾 error.

image

- [ ] On your README.md file, your mention that `ruby mylist.rb` will run the program. But you do not have such a file as `mylist.rb` in your project. As per the requirement, the `main.rb` file should be the entry point for your program. Kindly change this to the correct method of running your program.

**Your README instruction on Usage.**
image

The error message
image

File: get_service.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+require_relative './boolean'
+
+class GetSerivce
```
Comment (bobb-Rob) [inline get_service.rb:3]: - [ ] I see the error is from this class `GetSerivce`, perhaps the name is wrongly spelled in `app.rb` file. 
Maybe you should rename it to `GetService` and also correct it where you call this class in `app.rb` file.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+require 'date'
+require_relative './list_service'
+require_relative './get_service'
+require_relative './create_service'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @get_service = Get_serivce.new
```
Comment (bobb-Rob) [inline app.rb]: - [ ] This line of code should be 👇🏾 

```ruby
@get_service = GetService.new
```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @AhmedMohsen600 , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    else
+      puts ''
+    end
+  end
+
+  def main_section
+    'Please choose an option by entering a number:
+1 - List all books
+2 - List all people
+3 - Create a person
```
Comment (BenMukebo) [inline app.rb:57]: - [OPTIONAL] The `main.rb` file is the entry point that is invoked at the end of the file. You should Present the user with a list of options to perform here and not in the app.js file.
- This method should do and look similar to the following:

```ruby
def main
  app = App.new()

  response = nil

  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    ...
    puts "7 - Exit"
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
       ...
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

main()
```

**Review by BenMukebo — APPROVED** [review]
### Hi @AhmedMohsen600, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] When I click the link to the `MIT.md` file that you added, I get this 404 `error`. Please update your Readme, it will be nice to add and update this section(`License`) with the type of License and a link to the license file, as this will make your readme more informative and professional. Check out this [template](https://github.com/microverseinc/readme-template) for more clarifications



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**